### PR TITLE
Death plot added

### PR DESCRIPTION
**Pull request by njs59**

### CODE REVIEW

**Comment by jiayuanz3** [comment]
looks good to me. close https://github.com/SABS-R3-Epidemiology/epistrains/issues/66; close

File: epistrains/solver.py [hunk 1]
```diff
@@ -189,3 +189,43 @@ def save_compartments(self, save_path='epistrains_output.png'):
...
         plt = self._make_plot()
         plt.savefig(save_path, dpi=300)
+
+    def _make_death_plot(self):
+        """Creates the plot of the number of individuals
+        in each compartment over time
+        """
+        if self.solution is None:
+            raise ValueError("Must run s.solve() before plotting solutions")
+
```
Comment (HenrietteCapel) [inline epistrains/solver.py:199]: I would prefer making the cumulative line dark blue. And maybe also create a parameter for the colours to be consistent with the other plotting function.

File: epistrains/solver.py [hunk 2]
```diff
@@ -189,3 +189,43 @@ def save_compartments(self, save_path='epistrains_output.png'):
...
+        output_solver = self.solution
+
+        self._count_virus_death()
+        plt.plot(output_solver.t, self.deaths, label="Daily", color='brown')
+
+        plt.ylabel("Average number of deaths per day")
+        plt.xlabel("Time (days)")
+        ax = plt.gca()
+        ax2 = ax.twinx()
+        ax2.plot(output_solver.t, self.deaths.cumsum()/(len(output_solver.t)/(output_solver.t[-1]-output_solver.t[0])), label="Cumulative", color='b')
```
Comment (HenrietteCapel) [inline epistrains/solver.py:210]: Be consistent with "b" and "blue" in these two lines.

**Review by HenrietteCapel — COMMENTED** [review]
Some improvements desired.

**Comment by HenrietteCapel** [comment]
- Created test for the death plot
- created test for saving and plotting
- fixed return plt statement in make_plot (normal compartment plot)
- added comments
Need to: check flake8

File: epistrains/solver.py [hunk 3]
```diff
@@ -187,5 +188,50 @@ def save_compartments(self, save_path='epistrains_output.png'):
...
+        # plot daily deaths
+        self._count_virus_death()
+        plt.plot(output_solver.t, self.deaths, label="Daily", color=colours_deaths[1])
+
+        plt.ylabel("Average number of deaths per day")
+        plt.xlabel("Time (days)")
+        ax = plt.gca()
+        ax2 = ax.twinx()
+        # plot cumulative deaths
+        self.daily_cumulative_deaths = self.deaths.cumsum()/(len(output_solver.t)/(output_solver.t[-1]-output_solver.t[0]))
```
[the remark this answers is not in the document]
Comment (Ellmen) [reply epistrains/solver.py:214]: Or maybe we want to compute it in `_count_virus_deaths`? We could even call that at the end of `solve`

**Pull request merged**